stdlib/public/core/Reverse.swift

```diff
@@ -12,34 +12,47 @@
 
 /// An index that traverses the same positions as an underlying index,
 /// with inverted traversal direction.
-public struct ReverseIndex<Base> : Comparable {
-  public init(_ base: Base) { self.base = base }
```

Comment:
> The question in my mind is who else would want to create instances of these.

If you have an index into a collection at an interesting position, and you want to point at the same element in the reverse view.  Not a very common usecase, but still reasonable, I think.

Note that this does not apply to Filter directly, because not every index of the base collection is visible in the filtered collection.  So I'd be OK with not providing this API on Filter.  (A possible future extension though.)